NAAM.md

```diff
@@ -0,0 +1,282 @@
+single `EntryChunk` IPLD node, that contains a single multihash calculated as:
+
+* The SHA-256 multihash of IPNS Record Key, i.e. `/ipns/<ipns-key>`.
+   * The multihash value is encoded according to the IPNI advertisement encoding. See [IPNI Specification](IPNI.md). 
+   * There are no restrictions on the concrete IPNS key format as long as it complies with the [IPNS routing record specification](https://github.com/ipfs/specs/blob/main/ipns/IPNS.md#routing-record).
+
+This approach allows the resolvers to deterministically calculate the multihash used for lookup when
+interacting with the indexer node find APIs.
+
+#### Context ID
+
+The field `ContextID` in a regular advertisement is used as a way to uniquely identify the metadata
+associated to a list of multihashes. Naam uses a fixed value, `/ipni/naam`, as context ID to assure
+that the metadata associated to the published advertisement for the same IPNI key is singular and
+uniquely identifiable.
+
+Using a fixed value as context ID also enables network indexers to quickly differentiate Naam
+advertisements from a typical content advertisement. This opens up future opportunities for
+optimisation when it comes to ingesting Naam advertisements. For example, indexers can pro-actively
+process IPNS records and offer bespoke APIs that understand IPNS record expiry/validity.
+
+#### Metadata
+
+`Metadata` is typically used to convey the list of protocols over which the advertised content is
+retrievable. Further, this field is designed to be extensible: a valid metadata should start with a
+`varint` signifying _some_ protocol ID followed by arbitrary bytes. There are currently three well
+known protocol IDs. The protocol ID itself need not to be known by the indexer nodes. In fact,
+indexers treat metadata as a black box and carry on processing advertisements. Instead, it is meant
+to be a signal to the consumers of `Metadata`, i.e. retrieval clients, as a hint on how to decode
+the remaining bytes.
+
+Naam utilises this extensibility to directly encode IPNS records as metadata. The Naam advertisement
+metadata consists of:
+
+* `ipns` multicodec code `0xe5` as protocol ID,
+* followed by marshalled IPNS Record.
+
+The use of metadata to capture IPNS records enables Naam to also utilise the built-in IPNI mechanism
+for updating metadata to update IPNS records. Because, context ID nor the multihash in Naam
+advertisements ever change for the same IPNS key. For more information,
+see [Updating IPNS Records](#updating-ipns-records).
+
+Note that the IPNS multicodec code `0xe5` is tagged as "namespace" in the CSV table, which could
+signify an IPNS path, not record. There is currently no multicodec code for IPNS records themselves.
+Alternatively, a new multicodec code can be introduced to the
+[multicodec CSV file](https://github.com/multiformats/multicodec/blob/master/table.csv) to avoid
+potential confusion between the two. The new code can then be used as the Protocol ID of Naam
+Metadata.
```

Comment:
https://github.com/multiformats/multicodec/pull/312 merged, we no longer need this paragraph

```suggestion
```